velox/expression/ConstantExpr.cpp

```diff
@@ -72,4 +72,117 @@ std::string ConstantExpr::toString(bool /*recursive*/) const {
         "{}:{}", sharedSubexprValues_->toString(0), type()->toString());
   }
 }
+
+namespace {
+void appendSqlLiteral(
+    const BaseVector& vector,
+    vector_size_t row,
+    std::ostream& out);
+
+void appendSqlLiteralList(
+    const BaseVector& vector,
+    vector_size_t offset,
+    vector_size_t size,
+    std::ostream& out) {
+  for (auto i = offset; i < offset + size; ++i) {
+    if (i > 0) {
+      out << ", ";
+    }
+    appendSqlLiteral(vector, i, out);
+  }
+}
+
+void appendSqlString(const std::string& value, std::ostream& out) {
+  // Escape single quotes: ' -> ''.
+  static constexpr char kSingleQuote = '\'';
+
+  out << kSingleQuote;
+  auto prevPos = 0;
+  auto pos = value.find(kSingleQuote);
+  while (pos != std::string::npos) {
+    out << value.substr(prevPos, pos - prevPos) << kSingleQuote;
+    prevPos = pos;
+    pos = value.find(kSingleQuote, pos + 1);
+  }
+  out << value.substr(prevPos, pos) << kSingleQuote;
+  return;
+}
+
+void appendSqlLiteral(
+    const BaseVector& vector,
+    vector_size_t row,
+    std::ostream& out) {
+  if (vector.isNullAt(row)) {
+    out << "NULL";
+    return;
+  }
+
+  switch (vector.typeKind()) {
+    case TypeKind::BOOLEAN: {
+      auto value = vector.as<SimpleVector<bool>>()->valueAt(row);
+      out << (value ? "TRUE" : "FALSE");
+      break;
+    }
+    case TypeKind::TINYINT:
+    case TypeKind::SMALLINT:
+    case TypeKind::INTEGER:
+    case TypeKind::BIGINT:
+    case TypeKind::REAL:
+    case TypeKind::DOUBLE:
+      out << vector.wrappedVector()->toString(vector.wrappedIndex(row))
+          << "::" << vector.type()->toString();
+      break;
+    case TypeKind::VARCHAR: {
+      appendSqlString(
+          vector.wrappedVector()->toString(vector.wrappedIndex(row)), out);
+      break;
+    }
+    case TypeKind::ARRAY: {
+      out << "ARRAY[";
```

Comment:
Why only ARRAY is printed in caps and not map and row_constructor ? Should these be consistent ?

Reply:
map and row_constructor are function names, while ARRAY is a SQL keyword.